src/software/gui/robot_diagnostics/widgets/robot_diagnostics_gui.cpp

```diff
@@ -33,24 +33,162 @@ void RobotDiagnosticsGUI::onChickerStateChanged(double chicker_power,
                                                 ChargeMode charge_mode,
                                                 ChickMode chick_mode)
 {
-    // TODO (Issue #1420): push primitive set msg to buffer
+    if (chick_mode == ChickMode::CHIP)
+    {
+        chip_pressed = true;
+    }
+    else if (chick_mode == ChickMode::KICK)
+    {
+        kick_pressed = true;
+    }
+    auto direct_control_primitive = createDirectControlPrimitiveFromUI();
+    pushPrimitiveSetToBuffer(std::move(direct_control_primitive));
```

Comment:
So this seems like we're going with an "event-triggered update system" here. I think we should consider continually calling

```
auto direct_control_primitive = createDirectControlPrimitiveFromUI();
pushPrimitiveSetToBuffer(std::move(direct_control_primitive));
```

in a separate loop / timer, rather than _only_ when there is an update. The reason is that our communication (radio or wifi) is _unreliable communication_ (like UDP). This means there's a chance the message doesn't make it to the robot, and it won't be obvious to the user why the robot isn't doing anything when they clicked a button.

So it may be better to continually send primitives from the GUI.

Reply:
added